subsys/bluetooth/controller/ll_sw/ull_iso_internal.h

```diff
@@ -4,21 +4,43 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#if defined(CONFIG_BT_CTLR_ADV_ISO) || defined(CONFIG_BT_CTLR_CONN_ISO)
 /* Calculate ISO PDU buffers required considering SDU fragmentation */
-/* FIXME: Calculation considering both Connected and Broadcast ISO PDU
- *        fragmentation.
+#if defined(CONFIG_BT_CTLR_ADV_ISO) || defined(CONFIG_BT_CTLR_CONN_ISO)
+/* Internal ISO Tx SDU maximum length.
+ * A length that is minimum of the resultant combination of the HCI ISO data fragments provided and
+ * the user configured maximum transmit SDU length (CONFIG_BT_CTLR_ISO_TX_SDU_LEN_MAX).
+ */
+#define BT_CTLR_ISO_TX_SDU_LEN_MAX MIN(((CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE * \
+					 CONFIG_BT_CTLR_ISO_TX_BUFFERS) - \
+					BT_HCI_ISO_SDU_HDR_SIZE), \
+				       CONFIG_BT_CTLR_ISO_TX_SDU_LEN_MAX)
```

Comment:
Should we have some check if `CONFIG_BT_CTLR_ISO_TX_SDU_LEN_MAX < CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE - BT_HCI_ISO_SDU_HDR_SIZE`? It doesn't make sense to only support SDUs smaller than the HCI ISO data packet size

Reply:
Sure, BUILD_ASSERT or should we re-define `CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE` and allocate lower memory?.... BUILD_ASSERT may be better for user awareness than magically change the allocations?

Reply:
Yeah. Generally if I as a user set `CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE=100` and then get e.g. `80` in my buffers, then I'd be confused :D